src/SprykerEco/Zed/Braintree/Communication/Plugin/Checkout/BraintreeCheckoutPreConditionPlugin.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/**
+ * MIT License
+ * For full license information, please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace SprykerEco\Zed\Braintree\Communication\Plugin\Checkout;
+
+use Generated\Shared\Transfer\CheckoutResponseTransfer;
+use Generated\Shared\Transfer\QuoteTransfer;
+use Spryker\Zed\CheckoutExtension\Dependency\Plugin\CheckoutPreConditionPluginInterface;
+use Spryker\Zed\Kernel\Communication\AbstractPlugin;
+
+/**
+ * @method \SprykerEco\Zed\Braintree\Business\BraintreeFacadeInterface getFacade()
+ * @method \SprykerEco\Zed\Braintree\BraintreeConfig getConfig()
+ * @method \SprykerEco\Zed\Braintree\Persistence\BraintreeQueryContainerInterface getQueryContainer()
+ * @method \SprykerEco\Zed\Braintree\Communication\BraintreeCommunicationFactory getFactory()
+ */
+class BraintreeCheckoutPreConditionPlugin extends AbstractPlugin implements CheckoutPreConditionPluginInterface
+{
+    /**
+     * {@inheritDoc}
+     * - Requires `QuoteTransfer.payment.braintree` to be set.
+     * - Checks if quote payment is valid.
+     * - Returns `true` if payment provider is not Braintree.
+     * - Returns "false" and adds error to response transfer when condition fails.
+     * - Returns "true" when condition passed. Can add errors to response transfer.
+     * - Checks if Braintree has a nonce.
+     * - Checks a condition before the order is saved.
+     * - Does not change Quote transfer.
+     * - Does not write to Persistence.
```

Comment:
do we really need this `does not write...`?
it does not do many things

Reply:
Removed